### PR TITLE
Fix unreachable in block statement

### LINKED ISSUE #2187: compiler crash with unreachable in BlockStatement
code: 
```typescript
export function test(a: i32): i32 {
  let t = [];
  if (t.length == 0) {
    return 2;
  }
  {
    return unreachable();
  }
}
```

Error Message:

```
[wasm-validator error in function assembly/index/test] unexpected false: local.get must have a valid type - check what you provided when you constructed the node, on 
(local.get $4)
[wasm-validator error in function assembly/index/test] unexpected false: vars must be defaultable, on 
unreachable
FAILURE validate error
    at Object.main (/home/ccc/Desktop/test/as/node_modules/assemblyscript/cli/asc.js:953:23)
    at /home/ccc/Desktop/test/as/node_modules/assemblyscript/bin/asc:19:47
```

### PR DESCRIPTION
**Pull request by HerrCai0907**
⯈fix #2187
⯈support `return unreachable()`
⯈avoid compile wat like (return (unreachable)) and then insert shadowstack instrument like (locals.get (unreachable))

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: src/passes/shadowstack.ts [hunk 1]
```diff
@@ -658,6 +658,7 @@ class InstrumentReturns extends Pass {
     var stmts = new Array<ExpressionRef>();
     if (value) {
       let returnType = _BinaryenExpressionGetType(value);
+      assert(returnType != TypeRef.Unreachable, "shadowstack pass cannot handle return unreachable");
```
Comment (dcodeIO) [inline src/passes/shadowstack.ts:661]: If the returned value is unreachable, code generated here cannot be reached anyhow it seems:
```ts
let t = unreachableExpr; // aborts execution
__stack_pointer += frameSize
return t
```
As such, wdyt of covering this issue in the shadow pass directly, say by, if `returnType` is unreachable, doing nothing:
```ts
// Nothing to do if execution terminates before a value is returned
if (returnType == TypeRef.Unreachable) return;
```
Then other code can remain as-is, relying on the assumption that an unreachable value can substitute a concrete value.

Comment (HerrCai0907) [reply src/passes/shadowstack.ts]: The new commit has fixed it 😄

Comment (dcodeIO) [reply src/passes/shadowstack.ts]: With the check now in the shadow stack pass, can the other changes be reverted? :)

Comment (HerrCai0907) [reply src/passes/shadowstack.ts]: Yes, testcase is passing after trying to revert other change. 😃

**Review by dcodeIO — APPROVED** [review]

**Pull request merged**